File: 策略模式(锦囊妙计)/Context.java
//三个妙计都要放到锦囊中,Context就是锦囊
public class Context{
    //构造函数你要使用哪个锦囊
    private IStrategy strategy = null;
    public Context(IStrategy _strategy){
        this.strategy = _strategy;
    }
    //使用计谋了,看我出招
    public void operate(){
        this.strategy.operate();
    }
}

File: 通用代码/策略模式(C++)/Context.cpp
#include "Context.h"
Context::Context(Strategy* strategy)
{
    this->_strategy = strategy;
}

Context::~Context()
{
    delete this->_strategy;
    this->_strategy = NULL;
}

void Context::doAnything()
{
    this->_strategy->doSomething();
}
